### PR TITLE
Docs and linting

### PR DESCRIPTION
**Pull request by TendTo**
Although it may seem like a huge pr, it is actually one purely for maintenance purposes. Most of the changes regards docstrings and the sphinx documentation.
There are a few """dangerous changes""" that could be checked.

### Changes to check
- use of `dataclass` [ commit 181d87ea43c808a9f58b8e251cb48e94a5f4286d ]
- changes in `db_backup_job` [ commit a53a7735d6302ed48ebf954da50b2ad2490245f3 ]
- add class in tests [ commit  a4a00f2d70307bca7d0807f2c31ce06329715a84 ]

### Docs changes
- renamed some variables
- made sure all line lengths are under the set limit
- add .pylintrc file
- update sphinx doc website, adding dark mode
- simplified dostrings a lot, removing both tipehinting and default value. They are all computed by sphinx, and are already stated in the code, so they are pretty useless for the developer looking directly at the code
- added linting to github action
- some typos

### Utility changes
- added devcontainer for VsCode

### CODE REVIEW

**Review by Helias — APPROVED** [review]

File: modules/data/data_reader.py [hunk 1]
```diff
@@ -1,17 +1,17 @@
 """Read data from files"""
 import os
 from telegram.utils.helpers import escape_markdown
-from modules.data.config import Config
+from .config import Config
 
 
 def get_abs_path(*root_file_path: str) -> str:
     r"""Get the abs path from the root directory of the project to the requested path
```
Comment (drendog) [inline modules/data/data_reader.py:8]: can we remove this string prefix?

File: modules/data/data_reader.py [hunk 2]
```diff
@@ -21,10 +21,10 @@ def read_file(*root_file_path: str) -> str:
     r"""Read the contens of the file
```
Comment (drendog) [inline modules/data/data_reader.py:21]: here too

**Review by drendog — APPROVED** [review]

**Review by drendog — APPROVED** [review]

File: .devcontainer/devcontainer.json [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+{
+	"name": "Python 3",
+	"build": {
+		"dockerfile": "Dockerfile",
+		"context": "..",
+		"args": { 
+			// Update 'VARIANT' to pick a Python version: 3, 3.10, 3.9, 3.8, 3.7, 3.6
+			// Append -bullseye or -buster to pin to an OS version.
+			// Use -bullseye variants on local on arm64/Apple Silicon.
+			"VARIANT": "3.9-buster"
```
Comment (dcariotti) [inline .devcontainer/devcontainer.json:12]: This `VARIANT` version shouldn't be equals to `ARG VARIANT="3.10-bullseye"` of `.devcontainer/Dockerfile`?

Comment (dcariotti) [reply .devcontainer/devcontainer.json:12]: Ok, it's used for a compatibility with dev and prod setups

**Pull request merged**